TombEngine/Objects/TR3/Entity/tr3_civvy.cpp

```diff
@@ -79,341 +119,484 @@ namespace TEN::Entities::Creatures::TR3
 		SetAnimation(item, CIVVY_ANIM_IDLE);
 	}
 
+
+	//In the future, if this function works, it could be taken to a more neutral place to be reused by other ally entities.
+	ItemInfo* CivvyFindNearTarget(short itemNumber, std::vector<GAME_OBJECT_ID>& excludedTargetList, int rangeDetection)
+	{
+		auto& item = g_Level.Items[itemNumber];
+		auto& creature = *GetCreatureInfo(&item);
+
+		ItemInfo* result = nullptr;
+
+		auto closetsDistance = FLT_MAX;
+		Vector3 distanceVector;
+		float distanceValue;
+		auto* targetCreature = ActiveCreatures[0];
```

Comment:
This is not required, you can have auto* directly at: `targetCreature = ActiveCreatures[i];`
ActiveCreatures is a std::vector, you can do `for (auto* currentCreature : ActiveCreatures)` which speed it up.

Reply:
also distanceVector is not used there.

Reply:
done